libs/coroutines/include/hpx/coroutines/thread_id_type.hpp

```diff
@@ -59,93 +46,76 @@ namespace threads {
             thrd_ = nullptr;
         }
 
-        friend constexpr bool operator==(
-            std::nullptr_t, thread_id_type const& rhs)
+        friend constexpr bool operator==(std::nullptr_t, thread_id const& rhs)
         {
             return nullptr == rhs.thrd_;
         }
 
-        friend constexpr bool operator!=(
-            std::nullptr_t, thread_id_type const& rhs)
+        friend constexpr bool operator!=(std::nullptr_t, thread_id const& rhs)
         {
             return nullptr != rhs.thrd_;
         }
 
-        friend constexpr bool operator==(
-            thread_id_type const& lhs, std::nullptr_t)
+        friend constexpr bool operator==(thread_id const& lhs, std::nullptr_t)
         {
             return nullptr == lhs.thrd_;
         }
 
-        friend constexpr bool operator!=(
-            thread_id_type const& lhs, std::nullptr_t)
+        friend constexpr bool operator!=(thread_id const& lhs, std::nullptr_t)
         {
             return nullptr != lhs.thrd_;
         }
 
         friend constexpr bool operator==(
-            thread_id_type const& lhs, thread_id_type const& rhs)
+            thread_id const& lhs, thread_id const& rhs)
         {
             return lhs.thrd_ == rhs.thrd_;
         }
 
         friend constexpr bool operator!=(
-            thread_id_type const& lhs, thread_id_type const& rhs)
+            thread_id const& lhs, thread_id const& rhs)
         {
             return lhs.thrd_ != rhs.thrd_;
         }
 
         friend HPX_CXX14_CONSTEXPR bool operator<(
-            thread_id_type const& lhs, thread_id_type const& rhs)
+            thread_id const& lhs, thread_id const& rhs)
         {
             return std::less<void const*>{}(lhs.thrd_, rhs.thrd_);
         }
 
         friend HPX_CXX14_CONSTEXPR bool operator>(
-            thread_id_type const& lhs, thread_id_type const& rhs)
+            thread_id const& lhs, thread_id const& rhs)
         {
             return std::less<void const*>{}(rhs.thrd_, lhs.thrd_);
         }
 
         friend HPX_CXX14_CONSTEXPR bool operator<=(
-            thread_id_type const& lhs, thread_id_type const& rhs)
+            thread_id const& lhs, thread_id const& rhs)
         {
             return !(rhs > lhs);
         }
 
         friend HPX_CXX14_CONSTEXPR bool operator>=(
-            thread_id_type const& lhs, thread_id_type const& rhs)
+            thread_id const& lhs, thread_id const& rhs)
         {
             return !(rhs < lhs);
         }
 
         template <typename Char, typename Traits>
         friend std::basic_ostream<Char, Traits>& operator<<(
-            std::basic_ostream<Char, Traits>& os, thread_id_type const& id)
+            std::basic_ostream<Char, Traits>& os, thread_id const& id)
         {
             os << id.get();
             return os;
         }
 
     private:
-        thread_data* thrd_;
+        void* thrd_;
```

Comment:
May I suggest to use a `private: using thread_id_repr = void*;` (or similar) to remove the `void*` from the API?